api/src/pcapi/core/offers/repository.py

```diff
@@ -764,18 +765,45 @@ def get_filtered_stocks(
     order_by: StocksOrderedBy = StocksOrderedBy.BEGINNING_DATETIME,
     order_by_desc: bool = False,
 ) -> flask_sqlalchemy.BaseQuery:
-    query = models.Stock.query.filter(
-        models.Stock.offerId == offer_id,
-        models.Stock.isSoftDeleted == False,
+    query = (
+        models.Stock.query.join(models.Offer)
+        .join(offerers_models.Venue)
+        .filter(
+            models.Stock.offerId == offer_id,
+            models.Stock.isSoftDeleted == False,
+        )
     )
     if price_category_id is not None:
         query = query.filter(models.Stock.priceCategoryId == price_category_id)
     if date is not None:
         query = query.filter(sa.cast(models.Stock.beginningDatetime, sa.Date) == date)
     if time is not None:
+        venue_timezone = models.Offer.query.get(offer_id).venue.timezone
+
+        # Create a datetime object from the time object
+        dt = datetime.datetime.combine(datetime.date.today(), time)
+        # Make it aware of UTC timezone
+        dt = dt.replace(tzinfo=timezone("UTC"))
+        # Convert to the venue's timezone
+        dt = dt.astimezone(timezone(venue_timezone))
+        # Extract the time
+        venue_time = dt.time()
```

Comment:
(détail) je trouve ça plus clair 
```suggestion
        venue_timezone = models.Offer.query.get(offer_id).venue.timezone
        dt = datetime.datetime.combine(datetime.datetime.today(), time)
        venue_timezone = pytz.timezone(venue_timezone)
        # Transform user time input into the venue timezone 
        venue_time = dt.replace(tzinfo=pytz.utc).astimezone(venue_timezone).time()
```